staging/src/k8s.io/kubectl/pkg/polymorphichelpers/interface.go

```diff
@@ -67,7 +67,7 @@ var MapBasedSelectorForObjectFn MapBasedSelectorForObjectFunc = mapBasedSelector
 
 // ProtocolsForObjectFunc will call the provided function on the protocols for the object,
 // return nil-map if no protocols for the object, or return an error.
-type ProtocolsForObjectFunc func(object runtime.Object) (map[string]string, error)
+type ProtocolsForObjectFunc func(object runtime.Object) (map[string][]string, error)
```

Comment:
Is this backward compatible?

Reply:
No, but I don't think this change will have any impact. Only kubectl will use this

Reply:
It is public and it looks like people are using it (via github search, but possibly more elsewhere)...
https://github.com/search?q=ProtocolsForObjectFunc&type=code

Can we leave the existing function as-is and create a new one with the new behavior?

Perhaps mark the old one as deprecated, with an explanation that it doesn't support multiple protocols per port, and that code should use the new function instead?